app/src/main/kotlin/org/open/smsforwarder/ui/feedback/FeedbackViewModel.kt

```diff
@@ -0,0 +1,58 @@
+package org.open.smsforwarder.ui.feedback
+
+import androidx.lifecycle.ViewModel
+import com.github.terrakok.cicerone.Router
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import org.open.smsforwarder.R
+import org.open.smsforwarder.data.repository.FeedbackRepository
+import org.open.smsforwarder.domain.usecase.ValidateEmailUseCase
+import org.open.smsforwarder.utils.Resources
+import javax.inject.Inject
+
+@HiltViewModel
+class FeedbackViewModel @Inject constructor(
+    private val router: Router,
+    private val feedbackRepository: FeedbackRepository,
+    private val validateEmailUseCase: ValidateEmailUseCase
+) : ViewModel() {
+
+    private var _viewState = MutableStateFlow(FeedbackState())
+    val viewState = _viewState.asStateFlow()
+
+    fun onBackClick() = router.exit()
+
+    fun onSubmitClick(
+        email: String,
+        body: String,
+        callback: (Boolean) -> Unit,
+    ) {
+        feedbackRepository.sendFeedback(email, body, callback)
+        router.exit()
+    }
+
+    fun onEmailChanged(email: String) {
+        val emailValidationResult = validateEmailUseCase.execute(email)
+        _viewState.update {
+            it.copy(
+                emailInput = email,
+                emailInputError = emailValidationResult.errorMessage
+            )
+        }
+    }
+
+    fun onBodyChanged(body: String) {
+        _viewState.update { state ->
```

Comment:
I thin k you can create a use case for that as well.
ValidateBlankUseCase like it is done in validateEmailUseCase or validatePhoneUseCase. It is common case to validate fields on empty or blank and this use cases can be combined in the future too.